src/TRestTrackViewerProcess.cxx

```diff
@@ -0,0 +1,189 @@
+/// documentation
+/// is offered to you by the REST community. Your HELP is needed to keep this
+/// code
+/// up to date. Your feedback will be worth to support this software, please
+/// report
+/// any problems/suggestions you may find while using it at [The REST Framework
+/// forum](http://ezpc10.unizar.es). You are welcome to contribute fixing typos,
+/// updating
+/// information or adding/proposing new contributions. See also our
+/// <a href="https://github.com/rest-for-physics/framework/blob/master/CONTRIBUTING.md">Contribution
+/// Guide</a>.
+///
+///
+///--------------------------------------------------------------------------
+///
+/// RESTsoft - Software for Rare Event Searches with TPCs
+///
+/// \class      TRestTrackViewerProcess
+/// \author     Juanan Garcia
+///
+/// <hr>
+///
+#include "TRestTrackViewerProcess.h"
+
+ClassImp(TRestTrackViewerProcess);
+
+///////////////////////////////////////////////
+/// \brief Default constructor
+///
+TRestTrackViewerProcess::TRestTrackViewerProcess() { Initialize(); }
+
+///////////////////////////////////////////////
+/// \brief Default destructor
+///
+TRestTrackViewerProcess::~TRestTrackViewerProcess() {}
+
+///////////////////////////////////////////////
+/// \brief Function to initialize input/output event members and define the
+/// section name
+///
+void TRestTrackViewerProcess::Initialize() {
+    SetSectionName(this->ClassName());
+    SetLibraryVersion(LIBRARY_VERSION);
+
+    fTrackEvent = nullptr;
+    fSingleThreadOnly = true;
+}
+
+///////////////////////////////////////////////
+/// \brief Process initialization. It creates the canvas available in
+/// TRestEventProcess
+///
+void TRestTrackViewerProcess::InitProcess() {
+    CreateCanvas();
+    if (!hCanvas && fDrawHits)
+        hCanvas = new TCanvas("hitsCanvas", "hitsCanvas", fCanvasSize.X(), fCanvasSize.Y());
+}
+
+///////////////////////////////////////////////
+/// \brief The main processing event function
+///
+TRestEvent* TRestTrackViewerProcess::ProcessEvent(TRestEvent* evInput) {
+    // no need for verbose copy now
+    fTrackEvent = (TRestTrackEvent*)evInput;
+
+    fCanvas->cd();
+    TPad* pad = fTrackEvent->DrawEvent();
+    if (fDrawOriginEnd) {
+        GetOriginEnd(fAnalysisTree, originGr, endGr, legOrEnd);
+        DrawOriginEnd(pad, originGr, endGr, legOrEnd);
+    }
+
+    fCanvas->cd();
+    pad->Draw();
+    fCanvas->Update();
+
+    if (!fDrawHits) return fTrackEvent;
+
+    hCanvas->cd();
+    TPad* hPad = fTrackEvent->DrawHits();
+    if (fDrawOriginEnd) {
+        DrawOriginEnd(hPad, originGr, endGr, legOrEnd);
+    }
+
+    hCanvas->cd();
+    hPad->Draw();
+    hCanvas->Update();
+
+    return fTrackEvent;
+}
+
+///////////////////////////////////////////////
+/// \brief Get origin and end of the track if trackLineAnalysis have
+/// been performed
+///
+void TRestTrackViewerProcess::GetOriginEnd(TRestAnalysisTree* anaTree, std::vector<TGraph*>& origin,
```

Comment:
Why it receives a pointer to `TRestAnalysisTree`? The process will always have access to the analysis tree, i.e. using `this->GetFullAnalysisTree()` or `this->GetAnalysisTree()`.

Reply:
Perhaps this is a method to be implemented at `TRestTrackEvent`, or event at a new class `TRestTrackPainter` that implements statics methods, then `TRestTrackEvent` could inherit from `TRestTrackPainter`.

Reply:
Note that the function is declared `static`, so it need a `TRestAnalysisTree` passed to the function. I prefer to use `static` functions in this case because it allows me to use elsewhere.

Reply:
I understand the benefit of using static functions, just that perhaps should be placed at a common class contextualizing it, such as `TRestTrackPainter`. For me, it makes sense to associate a `GetOriginEnd` to a `TrackEvent`, but not to the process.

Reply:
Or perhaps a `TRestTrackAnalysis` that contains a `GetOriginEnd` implementing an algorithm that extracts origin/end from the track, but not from the analysis tree. I would do the other way around. I would use the method to fill the analysis tree.

Reply:
I agree with you that most of the static methods should be implemented in a dedicated `class`, perhaps in this case would be better to use a `namespace`.

I like the idea to have different `namespace` such as `TRestTrackPainter` and `TRestTrackAnalysis`. However, I think that this approach should be more generic:
 - `TRestTrackPainter` should implement all the track drawing methods.
 - `TRestTrackAnalysis` should implement all the track analysis methods.

We should probably use a similar schema for all the libraries by defining dedicated `TRestXXXPainter` and `TRestXXXAnalysis` namespaces in every single library.

However, this approach would require several changes in the code that cannot be addressed in this PR. I can give a hand with the migration, but it would require a lot of work. Let me know what do you think.

Reply:
Yes, that sounds good. If someone is leading that, I will also give a hand with the migration. 

For this PR I think we need to chat online about the real need to request analysisTree variables in that method.

Reply:
I am working on that, `TRestTrackViewerProcess` can call a static function on `TRestTrackLineAnalysisProcess` to calculate the origin and the end of the track.

Later on we can migrate all the static functions to `TRestTrackPainter` and `TRestTrackAnalysis`, or do yo prefer to do in this PR just with few static functions mentioned before?

Reply:
Yes, that could be done later.

Reply:
Ok, I have created an issue about this topic

https://github.com/rest-for-physics/framework/issues/183